src/transformers/tokenization_utils_base.py

```diff
@@ -1388,35 +1648,43 @@ def save_pretrained(self, save_directory) -> Tuple[str]:
     @add_end_docstrings(
         ENCODE_KWARGS_DOCSTRING,
         """
-            **kwargs: passed to the `self.tokenize()` method.
-    """,
+            **kwargs: Passed along to the `.tokenize()` method.
+        """,
+        """
+        Returns:
+            :obj:`List[int]`, :obj:`torch.Tensor`, :obj:`tf.Tensor` or :obj:`np.ndarray`:
+            The tokenized ids of the text.
+        """,
     )
     def encode(
         self,
         text: Union[TextInput, PreTokenizedInput, EncodedInput],
         text_pair: Optional[Union[TextInput, PreTokenizedInput, EncodedInput]] = None,
         add_special_tokens: bool = True,
-        padding: Union[bool, str] = False,
-        truncation: Union[bool, str] = False,
+        padding: Union[bool, str, PaddingStrategy] = False,
+        truncation: Union[bool, str, TruncationStrategy] = False,
         max_length: Optional[int] = None,
         stride: int = 0,
         return_tensors: Optional[Union[str, TensorType]] = None,
         **kwargs
-    ):
+    ) -> List[int]:
         """
-        Converts a string in a sequence of ids (integer), using the tokenizer and vocabulary.
+        Converts a string to a sequence of ids (integer), using the tokenizer and vocabulary.
 
         Same as doing ``self.convert_tokens_to_ids(self.tokenize(text))``.
 
+        .. warning::
+            This method is deprecated, ``__call__`` should be used instead.
```

Comment:
`encode` is still useful if you want a simple encoding of a string (without all the outputs) ie. `str -> List[int]`
I think we can remove the deprecation message.